src/txn/txn_recover.c

```diff
@@ -925,60 +985,24 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
-     * 1. The connection is not read-only. A read-only connection expects that there shouldn't be
-     *    any changes that need to be done on the database other than reading.
-     * 2. The history store file was found in the metadata.
-     */
-    if (hs_exists && !F_ISSET(conn, WT_CONN_READONLY)) {
-        /* Start the eviction threads for rollback to stable if not already started. */
-        if (!eviction_started) {
-            WT_ERR(__wt_evict_create(session));
-            eviction_started = true;
-        }
 
-        /*
-         * Currently, rollback to stable only needs to make changes to tables that use timestamps.
-         * That is because eviction does not run in parallel with a checkpoint, so content that is
-         * written never uses transaction IDs newer than the checkpoint's transaction ID and thus
-         * never needs to be rolled back. Once eviction is allowed while a checkpoint is active, it
-         * will be necessary to take the page write generation number into account during rollback
-         * to stable. For example, a page with write generation 10 and txnid 20 is written in one
-         * checkpoint, and in the next restart a new page with write generation 30 and txnid 20 is
-         * written. The rollback to stable operation should only rollback the latest page changes
-         * solely based on the write generation numbers.
-         */
-        WT_ASSERT(session,
-          conn->txn_global.has_stable_timestamp == false &&
-            conn->txn_global.stable_timestamp == WT_TS_NONE);
-
-        /*
-         * Set the stable timestamp from recovery timestamp and process the trees for rollback to
-         * stable.
-         */
-        conn->txn_global.stable_timestamp = conn->txn_global.recovery_timestamp;
-        conn->txn_global.has_stable_timestamp = false;
-
-        if (conn->txn_global.recovery_timestamp != WT_TS_NONE)
-            conn->txn_global.has_stable_timestamp = true;
-
-        __wt_verbose(session, WT_VERB_RTS,
-          "Performing recovery rollback_to_stable with stable timestamp: %s and oldest timestamp: "
-          "%s",
-          __wt_timestamp_to_string(conn->txn_global.stable_timestamp, ts_string[0]),
-          __wt_timestamp_to_string(conn->txn_global.oldest_timestamp, ts_string[1]));
-
-        WT_ERR(__wt_rollback_to_stable(session, NULL, false));
-    } else if (do_checkpoint)
+    if (do_checkpoint || rts_executed)
         /*
          * Forcibly log a checkpoint so the next open is fast and keep the metadata up to date with
          * the checkpoint LSN and archiving.
          */
         WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
+    /* Initialize the connection's base write generation after rollback to stable. */
+    WT_ERR(__wt_metadata_init_base_write_gen(session));
+
+    /*
+     * Update the open dhandles write generations and base write generation with the connection's
```

Comment:
The main reason behind updating the dhandles again with new base write gen is due to reset the transactions that are written as part of the recovery checkpoint. How about describing the same here?

The recovery checkpoint writes the pages to disk with new write generation number that contains transaction ids that are needed to reset later. Todo that, we increment the connection base write gen and btree write gen both.